### ISSUE TITLE
Panucatt Re-Arm Not Connecting With USB After Compiling Marlin 2.0

### ISSUE THREAD

**Issue by trnmcd** [MarlinFirmware/Marlin#13014]
First of all I just want to say thank you to everyone who puts in the work to make Marlin possible. I've used the 8-bit marlin for a few printers, but I'm experimenting with Marlin 2.0 for a corexy CNC project I'm doing. I was able to connect to the Re-ARM through USB, but after compiling Marlin 2.0 using Atom and Platform IO, I can't get my Re-ARM to connect to my computer again. I followed this tutorial from Chris which was very good:
https://www.youtube.com/watch?v=H-c8UTg-EMU
I was able to compile and upload to the board but I couldn't see the eeprom file on the board. I experimented a bit with the serial port setting and uncommented #define EEPROM_SETTINGS as somebody in the comments suggested. 
Now I can't even get my computer to recognize the board. I tried formatting the micro sd card again but my computer still won't pull up the board. The power led on the board does come on however. I'm unsure if it is a hardware or firmware error (it sounds like hardware a bit), but panucatt has no support which is why I am flashing Marlin instead of smoothie. I tried two different computers, and installing the USB driver like panucatt suggests. 
http://smoothieware.org/windows-drivers
I'm running windows 10 but I was desperate so I figured it was worth a shot. 
Any help is greatly appreciated and I just want to say thanks to everyone for making Marlin 2.0 happen!!

I don't know how to upload the zip file (this is my first time using Github) but I tried. I'm very sorry if it does not work.
[configuration.zip](https://github.com/MarlinFirmware/Marlin/files/2797776/configuration.zip)

 
`

**Comment by kAdonis** [comment]
Things have changed since the video was recorded
The EEPROM is no longer saved to the SD card, so you won't find it there
The Firmware.bin file wich is uploaded to the SD card gets renamed to Firmware.cur when marlin is  successfully installed on the Re-Arm
You find the correct Windows-USB driver in Marlin/src/HAL/HAL_LPC1768/win_usb_driver
It could take a few minutes til Windows shows the SD Card in Explorer

**Comment by trnmcd** [comment]
Thanks kAdonis for the quick reply. Haha, I wish I knew that before I went and botched everything up. I was able to connect to the board and see it in the beginning, I'm not sure what changed. I let the board stay plugged into the computer for 5 or so minutes and it still didn't recognize it. I'm thinking most likely the board is bricked. It's just weird that it still turns on but won't connect.

**Comment by trnmcd** [comment]
Is there any way I can just write to the sd card without connecting the board to my computer?

**Comment by RudyBenner** [comment]
Yes, best thing to do is try it. No more than 32gig. FAT32 format.

**Comment by Ketchu13** [comment]
@trnmcd Did you do your tests, with the temperature probes connected to the Ramps / RE-ARM?
If Marlin goes into protection, access to RE-ARM is no longer possible via USB

**Comment by gloomyandy** [comment]
If you have an SD card reader/adaptor you can just insert the SD card into your computer and copy the firmware over. Then insert the card into your board and restart it. If you do not see the Marlin device and the SD card appear on your computer, reboot again. If that does not work check the sd card to see if the firmware.bin has been renamed firmware.cur. You might also want to try a different SD card, smaller, slower ones are often better.

**Comment by kAdonis** [comment]
You can find the firmware.bin file in your marlin folder in .pioenvs/LPC1768  after you compiled
http://marlinfw.org/docs/basics/install_rearm.htm

**Comment by trnmcd** [comment]
@Ketchu13 Thanks for the suggestion. I'm just working on the Re-ARM without the RAMPS. Though it does sound like a similar situation.

**Comment by trnmcd** [comment]
> If you have an SD card reader/adaptor you can just insert the SD card into your computer and copy the firmware over. Then insert the card into your board and restart it. If you do not see the Marlin device and the SD card appear on your computer, reboot again. If that does not work check the sd card to see if the firmware.bin has been renamed firmware.cur. You might also want to try a different SD card, smaller, slower ones are often better.

> You can find the firmware.bin file in your marlin folder in .pioenvs/LPC1768 after you compiled
http://marlinfw.org/docs/basics/install_rearm.htm

@gloomyandy @kAdonis I moved the firmware.bin to the sd card and then put the sd card back into the Re-ARM and my computer recognized it! Thanks for the help, I have a little hope now. When it opens the Re-ARM it has an all caps FIRMWARE Cursor file. Does that sound correct? Thanks for all of the help.

**Comment by boelle** [comment]
slighty related q, is it possible to expose the onboard sd card so it's seen as a drive on debian?

might even be the default but i dont know this

EDIT: of course it has to be mounted on debian first before it can be used as a drive

**Comment by gloomyandy** [comment]
@trnmcd Yes that is correct it actually calls it FIRMWARE.CUR which on Windows may show up as a cursor file.

@boelle Yes you need to mount it. There is probably a way to make it automount, but exactly how you do that may vary from one version of linux to another.

**Comment by boelle** [comment]
@gloomyandy so the default is that it's exposed as a drive? i thought it was a well hidden setting in the firmware :-)

**Comment by boelle** [comment]
ahh https://discourse.octoprint.org/t/how-to-access-files-on-the-printers-sd-card/6518/3?u=boelle

so i need to send an M22 to "expose" it and then mount it. Did i get it right that far?

and i guess there is an option to automatic execute the M22 on marlin boot?

**Comment by gloomyandy** [comment]
By default the drive is exposed via USB from Marlin. But Octoprint sends an M21 as part of its connect sequence. But I'm not sure this has anything to do with the original question or with Marlin really.

**Comment by boelle** [comment]
no its starts to substitute hijacking

right now eyeballing the posts before mine to see if i can see why and what

**Comment by boelle** [comment]
@trnmcd the others have answered most questions, but i would connect a thermistor, and thou i have not done it i think you can do that correctly, just to keep marlin from locking you out

another great help/tool would be a sd card reader so you dont need to connect directly

and depending on how your board is mounted the sd card could get dificult to get to, so an extender might also be worth considering

or if you read my post mount the drive that is exposed through a pi and share it on your private network

lots of ways to skin a cat :-D

**Comment by trnmcd** [comment]
Thanks to everyone who helped me with my issue, I was able to edit my config and upload it to the Re-ARM directly. I haven't tested it on my machine yet, but the issue that I opened here is resolved. 
@boelle That's an interesting idea. I think I should be good now, going to hook it up later and see if most things work (I figure it'll take awhile to get it fully functioning). I'll look into your post about the pi, I'm currently using octoprint on the printer building my new machine. Thanks again to everyone who helped!!
